src/component-library/TextLink/TextLink.tsx

```diff
@@ -20,9 +20,7 @@ type TextLinkProps = Props & NativeAttrs;
 // TODO: merge this with CTALink
 const TextLink = forwardRef<HTMLAnchorElement, TextLinkProps>(
   ({ color = 'primary', external, to, underlined, size, weight, icon, children, ...props }, ref): JSX.Element => {
-    const linkProps: TextLinkProps = external
-      ? { to: { pathname: to as string }, target: '_blank', rel: 'noreferrer' }
-      : { to };
+    const linkProps: TextLinkProps = external ? { to, target: '_blank', rel: 'noreferrer' } : { to };
```

Comment:
`typeof to === "string" ? { to: { pathname: to as string } } : to`

this would allow to 

`<TextLink to={link} />` and also <TextLink to={{search:..., pathname: ...}}/>`

Reply:
```
type Props = {
...
to: string | Pick<LinkProps, "to">
}
```